### PR TITLE
Lcam-515-adding integration test

### PR DESCRIPTION
**Pull request by loganathanMOJ**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/LCAM-515)

Describe what you did and why.

Adding integration test for Crime Evidence endpoints.

Before you ask people to review this PR:

- [x] Tests should be passing: `./gradlew test`
- [x] Github should not be reporting conflicts; you should have recently run `git rebase main`.
- [x] Avoid mixing whitespace changes with code changes in the same commit. These make diffs harder to read and conflicts more likely.
- [x] You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- [x] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: crime-evidence/src/main/java/uk/gov/justice/laa/crime/evidence/config/ResourceServerConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+    @Bean
+    public AccessDeniedHandler bearerTokenAccessDeniedHandler() {
+        return new BearerTokenAccessDeniedHandler();
+    }
+
+    @Bean
+    SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
+        http
+                .cors().and().csrf().disable()
```
[the remark this answers is not in the document]
Comment (loganathanMOJ) [reply crime-evidence/src/main/java/uk/gov/justice/laa/crime/evidence/config/ResourceServerConfiguration.java]: Fixed now

File: crime-evidence/src/main/java/uk/gov/justice/laa/crime/evidence/exception/CrimeEvidenceExceptionHandler.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+    private static ResponseEntity<ErrorDTO> buildErrorResponse(HttpStatus status, String errorMessage) {
+        return new ResponseEntity<>(ErrorDTO.builder().code(status.toString()).message(errorMessage).build(), status);
+    }
+
+    @ExceptionHandler(ValidationException.class)
+    public ResponseEntity<ErrorDTO> handleValidationError(ValidationException ex) {
+        log.error("ValidationException: ", ex);
+        return buildErrorResponse(HttpStatus.BAD_REQUEST, ex.getMessage());
+    }
```
Comment (AafAnsari) [inline crime-evidence/src/main/java/uk/gov/justice/laa/crime/evidence/exception/CrimeEvidenceExceptionHandler.java:23]: ValidationException is not used anywhere in the code, so it's not required here

File: crime-evidence/src/main/java/uk/gov/justice/laa/crime/evidence/exception/CrimeEvidenceExceptionHandler.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+    @ExceptionHandler(APIClientException.class)
+    public ResponseEntity<ErrorDTO> handleApiClientError(APIClientException ex) {
+        log.error("APIClientException: ", ex);
+        return buildErrorResponse(HttpStatus.INTERNAL_SERVER_ERROR, ex.getMessage());
+    }
+
+    @ExceptionHandler(CrimeEvidenceDataException.class)
+    public ResponseEntity<ErrorDTO> handleCCPDataException(CrimeEvidenceDataException ex) {
```
Comment (AafAnsari) [inline crime-evidence/src/main/java/uk/gov/justice/laa/crime/evidence/exception/CrimeEvidenceExceptionHandler.java:32]: `handleCCPDataException` can be renamed to `handleCrimeEvidenceDataException`

File: crime-evidence/src/main/java/uk/gov/justice/laa/crime/evidence/exception/CrimeEvidenceExceptionHandler.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        log.error("CCPDataException: ", ex);
```
Comment (AafAnsari) [inline crime-evidence/src/main/java/uk/gov/justice/laa/crime/evidence/exception/CrimeEvidenceExceptionHandler.java:33]: `CCPDataException` can be renamed to `CrimeEvidenceDataException`

File: crime-evidence/src/main/java/uk/gov/justice/laa/crime/evidence/exception/ValidationException.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
+package uk.gov.justice.laa.crime.evidence.exception;
+
+/**
+ * <class>ValidationException</class>
+ */
+public class ValidationException extends RuntimeException {
+
```
Comment (AafAnsari) [inline crime-evidence/src/main/java/uk/gov/justice/laa/crime/evidence/exception/ValidationException.java:7]: This exception class is not called. Can be removed

**Review by AafAnsari — CHANGES_REQUESTED** [review]

**Review by AafAnsari — APPROVED** [review]

**Pull request merged**